### PR TITLE
Added axis=no_reduction to IoU losses

### LINKED ISSUE #1273: IoU and GIoU losses should support a way to reduce over NO axes
A [duplicate issue](https://github.com/keras-team/keras/issues/17207) exists in the Keras repo.

This is required for YoloX and was originally covered in #1006 with an approach suggested by @LukeWood([here](https://github.com/keras-team/keras-cv/pull/1006#discussion_r1020450632)). 

However, this was removed as per a recent review by @tanzhenyu.

### PR DESCRIPTION
**Pull request by quantumalaviya**
# What does this PR do?

Fixes #1273 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by LukeWood — APPROVED** [review]
Thanks Sarvagya!
